src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -43,13 +43,16 @@ object TaskGroupBuilder extends StrictLogging {
     resourceMatch: ResourceMatcher.ResourceMatch,
     volumeMatchOption: Option[PersistentVolumeMatcher.VolumeMatch]
   ): (mesos.ExecutorInfo, mesos.TaskGroupInfo, Seq[Option[Int]]) = {
+    verifyResources(podDefinition, resourceMatch.resources)
+
     val allEndpoints = podDefinition.containers.flatMap(_.endpoints)
 
     val mesosNetworks = buildMesosNetworks(
       podDefinition.networks, allEndpoints, resourceMatch.hostPorts, config.mesosBridgeName)
 
     val executorInfo = computeExecutorInfo(
       podDefinition,
+      resourceMatch.resources,
```

Comment:
Ah, so this is the fix?

Reply:
The fix is how `scalarResource` is changed and how it is used now.